### ISSUE TITLE
queryParams changes doesn't trigger RouterNavigation action

### ISSUE THREAD

**Issue by goldenass** [ngxs/store#922]
##I'm submitting a...

<pre><code>
[x ] Regression
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request 
[ ] Other... Please describe:
</code></pre>

## Current behavior

Navigating to the same path changing only the queryParams doesn't trigger a RouteNavigation action anymore.

this.store.dispatch(new Navigate('/search',{query:'a'})); //--> triggers RouterNavigation action
this.store.dispatch(new Navigate('/search',{query:'ab'})); //--> doesen't trigger RouterNavigation action

## Expected behavior

Changing anything in the route a RouterNavigation action must be triggered.

this.store.dispatch(new Navigate('/search',{query:'a'})); //--> triggers RouterNavigation action
this.store.dispatch(new Navigate('/search',{query:'ab'})); //--> also triggers RouterNavigation action

## What is the motivation / use case for changing the behavior?

that breaks code written for previous versions

## Environment

<pre><code>
Libs:
- @angular/core version: 7.2.9
- @ngxs/store version: 3.4.3

Browser:
- [ x] Chrome (desktop) version XX
- [ x] Chrome (Android) version XX
- [x ] Chrome (iOS) version XX
- [ x] Firefox version XX
- [ x] Safari (desktop) version XX
- [x ] Safari (iOS) version XX
- [ x] IE version XX
- [ x] Edge version XX

**Comment by arturovt** [comment]
@goldenass oh, that's interesting! Would be great if you could create a repro on the stackblitz

**Comment by goldenass** [comment]
@arturovt 

Here you are:
https://stackblitz.com/edit/angular-hacwyr

If you change the search value, routing happens, but the eventCount remains 1

**Comment by mucbits** [comment]
Facing the same problem.
This was working on 3.4.2: [stackblitz](https://stackblitz.com/edit/ngxs-simple-iwdk8f)
Updating to 3.4.3 breaks it: [stackblitz](https://stackblitz.com/edit/ngxs-simple-s4h8qh)

**Comment by arturovt** [comment]
@goldenass thank you, I will investigate!

**Comment by arturovt** [comment]
@goldenass @mucbits 
Guys, this might be fixed by #924. Could you please check it and say if everything alright?
You will need to install the `@dev` version from npm to test.

**Comment by mucbits** [comment]
Not sure how to install the new dev version of the router-plugin.
"npm i @ngxs/router-plugin@dev" will result in "@ngxs/router-plugin@3.4.3-dev.master-0ce2822" being installed.

("npm i @ngxs/store@dev" results in "@ngxs/store@3.4.3-dev.master-dd2349f" as expected)

It seems that the router-plugin did not get updated.

**Comment by goldenass** [comment]
@arturovt sorry it isn't fixed to me. I just tested on my application and in @mucbits stackbliz

**Comment by mucbits** [comment]
@goldenass be aware that router-plugin@dev might still refer to the old version

**Comment by goldenass** [comment]
@mucbits thanks, we just wrote our answer at almost the same time. I happened to read yours after the refresh of the pubblication of mine :D

**Comment by arturovt** [comment]
@goldenass @mucbits 
Wait gentlemen, it's probablly not published yet, because I've installed the `@dev` version on my local machine and there are no changes :smile:

**Comment by arturovt** [comment]
@goldenass @mucbits 
Could you check again please? The latest published version should be `3.4.3-dev.master-0f1e084`

**Comment by mucbits** [comment]
Switching queryParams is working now, but there is a problem with accessing the router state from an action dispatched by a child component.

I've updated the stackblitz: https://stackblitz.com/edit/ngxs-simple-iwdk8f
With 3.4.2 it shows:
"/child"
"/child"

But now the action dispatched by the child component is executed before the router state has been put into the store. So, it's not possible anymore to access the router state with selectSnapshot inside that action. Resulting in this been shown:
"/child"
null

**Comment by arturovt** [comment]
@mucbits 
Umm, what about `ngOnInit`?

**Comment by mucbits** [comment]
Until now it was possible to do any initial dispatches directly inside the constructor.
With a delay it’s working again. It wouldn’t be much work to change that.

However, it was possible to do that inside a child state before, so it might pose a breaking change for other applications too.

**Comment by arturovt** [comment]
@mucbits 
I understood you. Probably need to move dispatching from the `NavigationEnd` hook to the `GuardsCheckEnd || ResolveEnd`

**Comment by pankajsays** [comment]
I am facing this issue with my app. Change in query string does not cause the selectors to update. Can this issue be fixed in the next release?

**Comment by fabioscala** [comment]
I actually have a similar issue, but when routing to the same component with a different path:
eg. the page is on `baseUrl/a/b/c` and navigates to `baseUrl/a/b` which both resolve to the same component, the internal navigation events are not triggered, the state/selectors thus not updated.

**Comment by markwhitfeld** [comment]
@fabioscala Thanks for reporting this scenario. I haven't come across it before but I'm glad it is resolved. We will make sure to add this to the current PR #952 as a test to ensure we guard against regression.
cc @arturovt

**Comment by markwhitfeld** [comment]
This should be fixed in the latest `@dev` version.
Please could you test it out and let us know if it fixes your problem?

**Comment by arturovt** [comment]
@goldenass @mucbits 
We would appreciate your regression feedback gentlemen

**Comment by mdiin** [comment]
@arturovt Chiming in here to say I was hit by this exact problem just now. The most recent @dev version seems to have fixed the issue.

**Comment by jibwa** [comment]
I also hit this same issue. The fix worked once I updated store and router to dev. Thanks everyone.

I didn't see the explicit workaround instructions in full so here is a quick summary 

```
npm i --save @ngxs/router-plugin@dev
npm install --save @ngxs/store@dev

```
The versions I'm running are

"@ngxs/router-plugin": "^3.4.3-dev.master-faeab77"
"@ngxs/store": "^3.4.3-dev.master-faeab77"

**Comment by splincode** [comment]
Released in 3.5.0
